### PR TITLE
feat(risk): drop US_DRIVER_LICENSE from Presidio scans (POC-52)

### PR DESCRIPTION
**Pull request by mfbx9da4**
Closes [POC-52](https://linear.app/speakeasy/issue/POC-52/feedback-review-pii-us-driver-license-matching-pattern).

## Why

Presidio's `US_DRIVER_LICENSE` recognizer is unusably broad. Its "weak" alternative `[A-Z][0-9]{1,12}` matches arbitrary one-letter-plus-digits substrings inside base64 hashes, URL paths, UUID fragments, and version strings. Context boosting then upgrades the score to 0.65 whenever the substring `lic` appears nearby — which it does in any text containing **"public"**, **"duplicate"**, **"policy"**, **"explicit"**, etc., because the default context matcher does *substring*, not whole-word, matching.

Reference: **[microsoft/presidio#1063](https://github.com/microsoft/presidio/issues/1063)** — open since April 2023, no upstream fix on the regex.

Reproduced locally on production data: an `m2` substring inside `sha512-0UPjpq0G+v2RyzxDbeM8ConLICmLjRgcwuy4...` scored 0.65 as a US driver license because the surrounding token contained `LIC`. Same shape across `public_production` chat logs.

## What changed

Before / After:

| | Before | After |
|---|---|---|
| `presidioEntityBlacklist` | filters `PERSON` only | also filters `US_DRIVER_LICENSE` |
| Dashboard catalog | offers US driver license as a selectable rule | entry removed |
| `Government Identifiers` category preset | includes `pii.us_driver_license` | removed |
| Existing policies that still pin `US_DRIVER_LICENSE` | sent to Presidio | scrubbed at the `AnalyzeBatch` boundary, no DB migration needed |

Description map (`presidio.go`) and SQL category CASE statements (`server/internal/risk/queries.sql`) intentionally untouched so legacy `pii.us_driver_license` findings still resolve cleanly in the UI.

## Test plan

- [x] `go test ./server/internal/background/activities/risk_analysis/...` passes
- [x] `go test ./server/internal/risk/categories/...` passes
- [x] `TestFilterEntitiesDropsBlacklistedTypes` asserts `US_DRIVER_LICENSE` is stripped alongside `PERSON`, from mixed inputs and from all-blacklisted inputs (covers the short-circuit branch)
- [ ] Smoke against staging — verify no new `pii.us_driver_license` risk events land after deploy

---
## Summary by cubic
Remove US driver license detection from Presidio scans to stop noisy false positives (addresses Linear POC-52: https://linear.app/speakeasy/issue/POC-52). The rule is hidden and not serialized on new policies; legacy `pii.us_driver_license` findings still display.

- **Bug Fixes**
  - Blacklist US_DRIVER_LICENSE in the server Presidio entity filter; short-circuit when only blacklisted types are provided.
  - Dashboard: add a `hidden` flag; hide the rule from the form, counts, bulk toggles, and the visible Detection Rules catalog; keep its id reserved in `BUILTIN_RULE_IDS`; do not serialize hidden rules into `presidioEntities` unless the edited policy already pins them; remove it from the Government Identifiers preset.
  - Add tests confirming PERSON and US_DRIVER_LICENSE are stripped and that nil/empty cases behave correctly; existing policies are scrubbed server-side, no DB migration needed.

<sup>Written for commit 21fa5d313b0f5e202e418be09232859d5636a9c3. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/speakeasy-api/gram/pull/3172?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Review by alx-xo — APPROVED** [review]

File: client/dashboard/src/pages/security/policy-data.ts [hunk 1]
```diff
@@ -23,6 +23,12 @@ export type DetectionRule = {
   id: string;
   title: string;
   source: "gitleaks" | "presidio" | "prompt_injection";
+  // When true the rule is kept in the catalog so legacy risk_results still
```
Comment (vishalg0wda) [inline client/dashboard/src/pages/security/policy-data.ts:26]: Might it not be feasible to simply scrub these risk_results? That might be preferred since its likely FPs no?

**Review by vishalg0wda — APPROVED** [review]

File: client/dashboard/src/pages/security/policy-data.ts (hunk 1 above)
Comment (mfbx9da4) [reply client/dashboard/src/pages/security/policy-data.ts:26]: Yes definitely but that is pending @dennnis-ez work

File: client/dashboard/src/pages/security/policy-data.ts [hunk 2]
```diff
@@ -1277,9 +1283,14 @@ export const DETECTION_RULES: Record<RuleCategory, DetectionRule[]> = {
...
     },
     {
+      // Kept for legacy risk_results display; hidden from the policy form.
+      // Upstream Presidio regex is unusably broad — see microsoft/presidio#1063
+      // and PR #3172. Server-side blacklist also drops it before any
+      // Presidio request.
       id: "pii.us_driver_license",
       title: "US state-issued driver license",
       source: "presidio",
+      hidden: true,
```
[the remark this answers is not in the document]
Comment (mfbx9da4) [reply client/dashboard/src/pages/security/policy-data.ts:1293]: Not actionable — `hidden` is filtered out before the catalog is rendered.

`DetectionRules.tsx` (the built-in catalog) reads `BUILTIN_RULES_BY_CATEGORY`, which is derived from `DETECTION_RULES` with `.filter((r) => !r.hidden)` at `detection-rules-data.ts:140`. So `pii.us_driver_license` does not show up in the catalog rows. The two intentional non-filtered consumers (`risk-utils.ts` title fallback, `BUILTIN_RULE_IDS` collision set) are documented as such — they keep the id reserved and resolvable for legacy findings without exposing it as a selectable rule.
